### PR TITLE
Add stress and wall model bcs, set flux to array in neumman

### PR DESCRIPTION
**Pull request by timofeymukha**
Cherry-pick from stress_formulation.

- Sets the flux in neumann_t to be an array instead of a scalar.
- Adds shear_stress_t, which wraps 2 neumann_t and a dirichlet_t and allows setting a given stress in the x direction. Sets 0 stress in z and 0 velocity in y.
- Adds a wall_model_bc_t, which descends from shear_stress_t and uses a wall model to compute the stress.

### CODE REVIEW

File: src/scalar/scalar_scheme.f90 [hunk 1]
```diff
@@ -225,7 +226,7 @@ subroutine scalar_scheme_add_bcs(this, zones, bc_labels)
     character(len=NEKO_MSH_MAX_ZLBL_LEN), intent(in) :: bc_labels(:)
     character(len=NEKO_MSH_MAX_ZLBL_LEN) :: bc_label
     integer :: i, j, bc_idx
-    real(kind=rp) :: dir_value, flux_value
+    real(kind=rp) :: dir_value, flux
     logical :: bc_exists
```
Comment (timfelle) [inline src/scalar/scalar_scheme.f90:230]: Some of these are unused.
```suggestion
    integer :: i
    real(kind=rp) :: dir_value, flux
```

File: src/.depends
Comment (timfelle) [inline src/.depends:1]: Please add `math.o` to `scalar_scheme.f90`

**Review by timfelle — APPROVED** [review]

File: src/scalar/scalar_scheme.f90 [hunk 2]
```diff
@@ -68,6 +68,7 @@ module scalar_scheme
   use utils, only : neko_error
   use comm, only: NEKO_COMM, MPI_INTEGER, MPI_SUM
   use scalar_source_term, only : scalar_source_term_t
+  use math, only : cfill
```
Comment (timfelle) [inline src/scalar/scalar_scheme.f90:71]: Imported but not used.

```suggestion
```

**Review by timfelle — APPROVED** [review]

File: src/scalar/scalar_scheme.f90 [hunk 3]
```diff
@@ -251,15 +251,16 @@ subroutine scalar_scheme_add_bcs(this, zones, bc_labels)
           call this%dir_bcs(this%n_dir_bcs)%mark_zone(zones(i))
           read(bc_label(3:), *) dir_value
           call this%dir_bcs(this%n_dir_bcs)%set_g(dir_value)
+          call this%dir_bcs(this%n_dir_bcs)%finalize()
 !          end if
```
Comment (njansson) [inline src/scalar/scalar_scheme.f90:255]:
```suggestion
```

Some left over comments

**Review by njansson — APPROVED** [review]

**Pull request merged**